google/cloud/aiplatform/utils/__init__.py

```diff
@@ -325,6 +325,34 @@ def extract_bucket_and_prefix_from_gcs_path(gcs_path: str) -> Tuple[str, Optiona
     return (gcs_bucket, gcs_blob_prefix)
 
 
+def extract_project_and_location_from_parent(
+    parent: str,
+) -> Dict[Optional[str], Optional[str]]:
+    """Given a complete parent resource name, return the project and location as a dict.
+
+    Example Usage:
+
+        parent_resources = extract_project_and_location_from_parent(
+            "projects/123/locations/us-central1/datasets/456"
+        )
+
+        parent_resources["project"] = "123"
+        parent_resources["location"] = "us-central1"
+
+    Args:
+        parent (str):
+            Required. A complete parent resource name.
+
+    Returns:
+        Dict[Optional[str], Optional[str]
+            A project, location dict from provided parent resource name.
+    """
+    parent_resources = re.match(
```

Comment:
Is this code basically copied from the GAPIC library such as this?  https://github.com/googleapis/python-aiplatform/blob/main/google/cloud/aiplatform_v1/services/dataset_service/client.py#L377

Why not just use that instead?

Reply:
When that code extracts the project and location from the string `"projects/123/locations/us-central1/datasets/456"` it extracts it as `{"project" : "123", "location" : "us-central1/datasets/456"}`, it takes everything remaining after `locations/`. My regex is based on the one you and Sasha sent, but the end parsing is different to only grab `us-central1`.

Reply:
I missed that - thanks for the explanation!